src/frontend/src/pages/SearchPage.jsx

```diff
@@ -50,117 +56,115 @@ export default function SearchPage() {
-      setTotalItems(response.numberOfItems || resultList.length);
-      applyFilters(resultList, filterCriteria);
-    } catch (error) {
-      console.error('Error searching:', error.message);
-      showToastMessage(
-        error.message || 'Error occurred while searching.',
-        'danger',
-      );
-      setResults([]);
-      setFilteredResults([]);
-    } finally {
-      setLoading(false);
-    }
-  }, [
-    query,
-    queryType,
-    filterCriteria,
-    keywords,
-    structuredFields,
-    currentMediaPage,
-    mediaItemsPerPage,
-    applyFilters,
-    showToastMessage,
-  ]);
-
-  const handleAdvancedSearch = useCallback((params) => {
-    if (params.query_type) setQueryType(params.query_type);
-
-    if (params.query_type === 'Simple' && params.query) {
-      setQuery(params.query);
-    }
-
-    if (
-      (params.query_type === 'ExactMatch' ||
-        params.query_type === 'BestMatch') &&
-      params.keywords
-    ) {
-      setKeywords(params.keywords);
-      setQuery('');
-    }
-
-    if (params.query_type === 'Structured') {
-      setStructuredFields({
-        title: params.title || '',
-        plot: params.plot || '',
-        character: params.character || '',
-        person: params.person || '',
-      });
-      setQuery('');
-    }
-
-    setCurrentMediaPage(1);
-  }, []);
-
-  const handleMediaPageChange = (pageNumber) => {
-    setCurrentMediaPage(pageNumber);
-  };
-
-  useEffect(() => {
-    handleSearch();
-  }, [
-    query,
-    queryType,
-    keywords,
-    structuredFields,
-    filterCriteria,
-    currentMediaPage,
-    handleSearch,
-  ]);
+    },
+    [
+      applyFilters,
+      filterCriteria,
+      mediaPage.count,
+      mediaPage.page,
+      personPage.count,
+      personPage.page,
+      showToastMessage,
+    ],
+  );
 
   useEffect(() => {
-    const searchParams = new URLSearchParams(location.search);
-    const queryValue = searchParams.get('q') || '';
-    setQuery(queryValue);
-  }, [location.search]);
+    performSearch(query);
+  }, [performSearch, query]);
+
+  let resultBody = <></>;
+  if (filteredResults.length === 0 || personResults.length === 0) {
+    resultBody = (
+      <div className="d-flex justify-content-center align-items-center py-5">
+        <h2 className="text-muted">
+          No matches found. Try a different search.
+        </h2>
+      </div>
+    );
+  } else {
+    resultBody = (
+      <>
+        <MediaGrid media={filteredResults} loading={loading} />
+        {filteredResults.length > 0 &&
+          personResults.numberOfItems > personPage.count && (
+            <Pagination
+              totalItems={mediaResults.numberOfItems}
+              itemsPerPage={mediaPage.count}
+              currentPage={mediaPage.page}
+              onPageChange={(pageNumber) =>
+                setMediaPage({ ...mediaPage, page: pageNumber })
+              }
+            />
+          )}
+        <PersonsGrid persons={personResults.items} />
```

Comment:
It would be nice I think if we had a headline for the persons as in the design: https://www.figma.com/design/R5qjDjDpJqpTQ18iX2LhF8/Wireframe-portfolio-3?node-id=1-962&t=u8ZbENA5wGJD29IZ-4